packages/playwright/src/runner/loadUtils.ts

```diff
@@ -118,7 +121,50 @@ export async function loadFileSuites(testRun: TestRun, mode: 'out-of-process' |
   }
 }
 
-export async function createRootSuite(testRun: TestRun, errors: TestError[], shouldFilterOnly: boolean): Promise<Suite> {
+export async function detectChangedFiles(testRun: TestRun): Promise<string[]> {
+  const baseCommit = testRun.config.cliOnlyChanged;
+
+  function gitFileList(command: string) {
+    try {
+      return childProcess.execSync(
+          `git ${command}`,
+          { encoding: 'utf-8', stdio: 'pipe' }
+      ).split('\n').filter(Boolean);
+    } catch (_error) {
+      const error = _error as childProcess.SpawnSyncReturns<string>;
+      throw new Error([
```

Comment:
I'd print something like

```
Cannot detect changed files for --only-changed mode:
<git command executed>

<git output>
```

Reply:
done in https://github.com/microsoft/playwright/pull/31727/commits/b6ef01324fa3dd0dc6bf70d348bb38d409ce7c55